openff/interchange/smirnoff/_create.py

```diff
@@ -93,17 +97,61 @@ def validate_topology(value):
         )
 
 
+def _preprocess_preset_charges(
+    molecules_with_preset_charges: list[Molecule] | None,
+) -> list[Molecule] | None:
+    """
+    Pre-process the molecules_with_preset_charges argument.
+
+    If molecules_with_preset_charges is None, return None.
+
+    If molecules_with_preset_charges is list[Molecule], ensure that
+
+    1. The input is a list of Molecules
+    2. Each molecule has assign partial charges
+    3. Ensure no molecules are isomorphic with another in the list
+
+    """
+    if molecules_with_preset_charges is None:
+        return None
+
+    molecule_set = {molecule.to_smiles() for molecule in molecules_with_preset_charges}
```

Comment:
```suggestion
    molecule_set = {molecule for molecule in molecules_with_preset_charges}
```

`Molecule.__eq__()` exists, so you should be able to make a set out of molecules directly. It definitely seems to work for me. Is matching SMILES faster for large topologies or something?

Reply:
I figure making SMILES is probably slower because of the short-circuits in `are_isomorphic`. I think relying on built-in equality would work fine here, but I also want to encode the intentionality of uniqueness being defined here in a way that's different than how the toolkit does it, which may change. (I'm not sure `.to_smiles()` is the best way to do this

I'm pretty neutral on this change, but your suggestion is simpler. I'll go with it and note the context in a comment